packages/react-components/src/trace-explorer/trace-explorer-views-widget.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import { List, ListRowProps, AutoSizer } from 'react-virtualized';
+import { OutputAddedSignalPayload } from '@trace-viewer/base/lib/signals/output-added-signal-payload';
+import { signalManager, Signals } from '@trace-viewer/base/lib/signals/signal-manager';
+import { OutputDescriptor } from 'tsp-typescript-client/lib/models/output-descriptor';
+import { Experiment } from 'tsp-typescript-client/lib/models/experiment';
+import { AvailableViewsChangedSignalPayload } from '@trace-viewer/base/lib/signals/available-views-changed-signal-payload';
+// import { ExperimentManager } from '@trace-viewer/base/src/experiment-manager';
+
+export interface ReactAvailableViewsProps {
+    id: string,
+    title: string,
+    // experimentManager: ExperimentManager,
+    contextMenuRenderer?: (anchor: {x: number, y: number}, output: OutputDescriptor) => void,
+}
+
+export interface ReactAvailableViewsState {
+    availableOutputDescriptors: OutputDescriptor[];
+}
+
+export class ReactAvailableViewsWidget extends React.Component<ReactAvailableViewsProps, ReactAvailableViewsState> {
+    static LIST_MARGIN = 2;
+    static LINE_HEIGHT = 16;
+    static ROW_HEIGHT = (2 * ReactAvailableViewsWidget.LINE_HEIGHT) + ReactAvailableViewsWidget.LIST_MARGIN;
+
+    protected forceUpdateKey = false;
+    protected lastSelectedOutputIndex = -1;
+
+    protected selectedExperiment: Experiment | undefined;
+
+    constructor(props: ReactAvailableViewsProps) {
+        super(props);
+        signalManager().on(Signals.AVAILABLE_OUTPUTS_CHANGED, (payload: AvailableViewsChangedSignalPayload) => this.doHandleAvailableViewsChanged(payload));
+        this.state = { availableOutputDescriptors: [] };
+    }
+
+    componentWillUnmount(): void {
+        signalManager().off(Signals.AVAILABLE_OUTPUTS_CHANGED, (payload: AvailableViewsChangedSignalPayload) => this.doHandleAvailableViewsChanged(payload));
```

Comment:
The listener does not get removed because it is not the same instance being passed that was added earlier.